src/coredump/coredump.c

```diff
@@ -1119,9 +1214,23 @@ static int process_kernel(int argc, char* argv[]) {
 
         if (get_process_root(pid, &t) >= 0) {
                 core_root = strjoina("COREDUMP_ROOT=", t);
-                free(t);
 
                 IOVEC_SET_STRING(iovec[n_iovec++], core_root);
+
+                /* If the process' root is "/", then there is a chance it has
+                 * mounted own root and hence being containerized. */
+                if (strcmp(t, "/") == 0) {
```

Comment:
Good, I was thinking about this construction but I hesitated to introduce the new variable - mainly because I must declare the variable at the beginning of the function and I am afraid of introducing single purpose variables ~200 lines far from their usage.